sw/device/tests/sim_dv/pattgen_ios_test.c

```diff
@@ -0,0 +1,189 @@
+// Copyright lowRISC contributors.
+// Licensed under the Apache License, Version 2.0, see LICENSE for details.
+// SPDX-License-Identifier: Apache-2.0
+
+#include "sw/device/lib/base/mmio.h"
+#include "sw/device/lib/dif/dif_pattgen.h"
+#include "sw/device/lib/dif/dif_pinmux.h"
+#include "sw/device/lib/runtime/hart.h"
+#include "sw/device/lib/runtime/irq.h"
+#include "sw/device/lib/runtime/log.h"
+#include "sw/device/lib/testing/rv_plic_testutils.h"
+#include "sw/device/lib/testing/test_framework/check.h"
+#include "sw/device/lib/testing/test_framework/ottf_main.h"
+
+#include "hw/top_earlgrey/sw/autogen/top_earlgrey.h"
+
+/**
+   PATTGEN IOS test
+   The test programs both channels of the pattgen to random patterns.
+   Pattgen ios are connected as follows:
+   pda_tx[0] - IOB9
+   pcl_tx[0] - IOB10
+   pda_tx[1] - IOB11
+   pcl_tx[1] - IOB12
+
+   The random pattern programmed by 'chip_sw_patt_ios_vseq.sv'
+   and output will be collected at the PADS and compared in SV test bench.
+ */
+
+OTTF_DEFINE_TEST_CONFIG();
+
+// 1: channel 0, 2: channel 1, 3: Both channels
+static volatile const uint8_t kChannelEnable = 3;
```

Comment:
Maybe add a comment saying the following volatile values will be override by SV testbench with random values?

Reply:
sure